File: delivery-info-service/src/main/java/com/potatorider/repository/DeliveryRepository.java
package com.potatorider.repository;

import com.potatorider.domain.Delivery;

import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import reactor.core.publisher.Flux;

public interface DeliveryRepository extends ReactiveMongoRepository<Delivery, String> {

    Flux<Delivery> findAllBy(Pageable pageable);
}

File: delivery-info-service/src/main/java/com/potatorider/repository/DeliveryRepository.kt
package com.potatorider.repository

import com.potatorider.domain.Delivery
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface DeliveryRepository : ReactiveMongoRepository<Delivery, String> {
    fun findAllBy(pageable: Pageable): Flux<Delivery>
}
